### ISSUE TITLE
@requires directive

### ISSUE THREAD

**Issue by tazsingh** [0xR/graphql-transform-federation#2]
Hey @0xR, thanks a bunch for this library. It's proven very helpful!

I saw your [note in the README about the `@requires` directive](https://github.com/0xR/graphql-transform-federation#the-requires-directive) and thought I'd provide a use-case that we have as an example of where it's needed. I do agree that if you have an existing graph it may not be overly useful. However, Federation is meant to glue together multiple graphs and their backing services where it does become quite useful.

I'm using `graphql-transform-federation` for a new graph that's implemented with the `nexus` library. Since we have different backing Federated services, some of which are wrappers around 3rd party APIs that our front-ends consume, we often need a myriad of fields from other services to satisfy the resolution of a field on an individual service. For example, we have the following definition in our Federated Graph at the moment for a wrapper around a 3rd party API (fields and types have been renamed to obfuscate our direct use-case):

```graphql
extend type Account @key(fields: "id") {
  id: ID! @external

  # This ID is stored in a different service and is pulled into this service
  # which would then talk to that external service.
  externalSystemId: String @external
  primaryContact: User @external
  accountType: [AccountType!] @external

  # This field `@requires` a variety of other fields, to then pass off to
  # a 3rd party service, in order for that service to have enough
  # information to upsert the entity for us.
  accountExternalProfile: AccountExternalProfile
    @requires(
      fields: "externalSystemId accountType primaryContact { id email }"
    )

  # This field will actually make an API call to the externally wrapped service
  isValidExternalAccount: Boolean @requires(fields: "accountType")

  # Here we'd only have this field for particular account types
  externalSystemAccountId: String @requires(fields: "accountType")
}
```

I hope I've done a good enough job of providing an example of where the `@requires` directive is useful. Let me know if I can be of further assistance in resolving any ambiguities or with the implementation of the feature itself.

Thanks again!

**Comment by 0xR** [comment]
Released in 2.1.0

**Comment by tazsingh** [comment]
Thanks @0xR for merging @jgnieuwhof's PR 😄 (We're on the same team ❤️)
